File: src/main/java/net/ccbluex/liquidbounce/ui/client/clickgui/style/styles/fdpdropdown/utils/animations/impl/EaseInOutQuad.java
/*
 * FDPClient Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/SkidderMC/FDPClient/
 */
package net.ccbluex.liquidbounce.ui.client.clickgui.style.styles.fdpdropdown.utils.animations.impl;

import net.ccbluex.liquidbounce.ui.client.clickgui.style.styles.fdpdropdown.utils.animations.Animation;
import net.ccbluex.liquidbounce.ui.client.clickgui.style.styles.fdpdropdown.utils.animations.Direction;

public class EaseInOutQuad extends Animation {

    public EaseInOutQuad(int ms, double endPoint, Direction direction) {
        super(ms, endPoint, direction);
    }

    protected double getEquation(double x1) {
        double x = x1 / getDuration();
        return x < 0.5 ? 2 * Math.pow(x, 2) : 1 - Math.pow(-2 * x + 2, 2) / 2;
    }

}

File: src/main/java/net/ccbluex/liquidbounce/ui/client/clickgui/style/styles/nlclickgui/animations/impl/EaseInOutQuad.kt
package net.ccbluex.liquidbounce.ui.client.clickgui.style.styles.nlclickgui.animations.impl

import net.ccbluex.liquidbounce.ui.client.clickgui.style.styles.nlclickgui.animations.Animation
import net.ccbluex.liquidbounce.ui.client.clickgui.style.styles.nlclickgui.animations.Direction

class EaseInOutQuad @JvmOverloads constructor(
    ms: Int,
    endPoint: Double,
    direction: Direction = Direction.FORWARDS
) : Animation(ms, endPoint, direction) {

    override fun getEquation(x1: Double): Double {
        val x = x1 / duration
        return if (x < 0.5) 2 * Math.pow(x, 2.0) else 1 - Math.pow(-2 * x + 2, 2.0) / 2
    }
}
